src/routers/StackRouter.js

```diff
@@ -521,7 +521,15 @@ export default (routeConfigs, stackConfig = {}) => {
         }
         const nextResult = result || {};
         const paramName = key.name;
-        nextResult[paramName] = matchResult;
+
+        let decodedMatchResult;
+        try {
+          decodedMatchResult = decodeURIComponent(matchResult);
```

Comment:
is there any way this could be decoded to empty string (other than empty string), which would then cause `decodedMatchResult || matchResult` to return `matchResult`?

Reply:
No, I think.
When it has a null character, `decodeURIComponent('%00')` returns `'\u0000'`.